jerry-core/api/jerry.c

```diff
@@ -6426,6 +6427,135 @@ jerry_get_container_type (const jerry_value_t value) /**< the container object *
   return JERRY_CONTAINER_TYPE_INVALID;
 } /* jerry_get_container_type */
 
+/**
+ * Return a new array containing elements from a Container or a Container Iterator.
+ * Sets the boolean input value to `true` if the container object is a
+ * key-value structure and `false` if not.
+ *
+ * Note:
+ *     the returned value must be freed with a jerry_release_value call
+ *
+ * @return jerry_value_t representing an array containing the elemets of the map.
+ *         jerry_value_t representing an empty array, if the Iterator is finished/empty.
+ *         Error if the value is not a Container or Container Iterator.
+ *         `undefined` if the `value` is undefined or null.
+ */
+jerry_value_t
+jerry_get_array_from_container (jerry_value_t value, /**< the container or iterator object */
+                                bool *is_key_value_p) /**< [out] is key-value structure */
+{
+  jerry_assert_api_available ();
+
+  if (jerry_value_is_undefined (value) || jerry_value_is_null (value))
+  {
+    return jerry_create_undefined ();
+  }
+
+#if JERRY_BUILTIN_CONTAINER
+  if (!ecma_is_value_object (value))
+  {
+    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG ("Value is not a Container or Iterator")));
+  }
+
+  ecma_collection_t *collection_buffer = ecma_new_collection ();
+
+  ecma_object_t *obj_p = ecma_get_object_from_value (value);
+  ecma_extended_object_t *ext_obj_p = (ecma_extended_object_t *) obj_p;
+  ecma_object_type_t type = ecma_get_object_type (obj_p);
+
+  ecma_collection_t *container_p;
+  uint32_t entry_count;
+  uint8_t entry_size;
+
+  uint32_t index = 0;
+  uint8_t iterator_kind = ECMA_ITERATOR__COUNT;
+  ecma_value_t *start_p;
+
+  *is_key_value_p = false;
+
+  if (type == ECMA_OBJECT_TYPE_CLASS
```

Comment:
maps/sets also has this type, we could return early if this check fails. There is no need a `type` variable then in the code.